src/CatalogService.Core/Entities/Item.cs

```diff
@@ -1,5 +1,139 @@
-namespace CatalogService.Core.Entities;
-
-public class Item {
-    
+using CatalogService.Core.Exceptions;
+using CatalogService.Core.Helpers;
+using CatalogService.Core.Interfaces;
+using CatalogService.Core.Models;
+using FluentValidation;
+using OneOf;
+using OneOf.Types;
+
+namespace CatalogService.Core.Entities;
+
+public class Item
+{
+    private readonly IValidator<ItemModel> _validator;
+    private readonly IItemRepository _repository;
+    private readonly CancellationToken _cancellationToken;
+
+    public Item(IValidator<ItemModel> validator, IItemRepository repository, CancellationToken cancellationToken = default)
+    {
+        _validator = validator;
+        _repository = repository;
+        _cancellationToken = cancellationToken;
+    }
+
+    public async Task<OneOf<long, ItemFailureException>> Add(ItemModel item)
+    {
+        var validate = _validator.Validate(item);
+
+        if (!validate.IsValid)
+            return new ItemFailureException(validate.Errors);
+
+        try
+        {
+            return await _repository.Add(item, _cancellationToken);
+        }
+        catch (Exception ex)
+        {
+            return new ItemFailureException(ex.Message);
+        }
+    }
+
+    public async Task<OneOf<Success, NotFound, ItemFailureException>> Update(ItemModel item)
+    {
+        try
+        {
+            var response = await GetById(item.Id);
+            var (foundItem, itemReturned) = WasItemFound(response);
+
+            if (!foundItem)
+                return new NotFound();
+
+            item = MapChangedFields(item, itemReturned!);
+
+            item = item with
+            {
+                Description = string.IsNullOrEmpty(item.Description) ? null : item.Description,
+                Image = string.IsNullOrEmpty(item.Image) ? null : item.Image,
+            };
+
+            var validate = _validator.Validate(item);
+
+            if (!validate.IsValid)
+                return new ItemFailureException(validate.Errors);
+
+            await _repository.Update(
+                item with { Id = itemReturned!.Id }, 
+                _cancellationToken);
+        }
+        catch (Exception ex)
+        {
+            return new ItemFailureException(ex.Message);
+        }
+
+        return new Success();
+    }
+
+    private ItemModel MapChangedFields(ItemModel newValue, ItemModel existent) =>

```

Comment:
Not in this case, I put this method in here due to the issue that when updating an Item I can partially send the body, a hash here would only save me the time if the objects were identical so I wouldn't need to update it at all. Other way to implement it would be on database level.